crates/services/p2p/src/peer_report.rs

```diff
@@ -71,15 +84,17 @@ pub struct Behaviour {
     connected_reserved_nodes: HashSet<PeerId>,
     pending_connections: HashSet<ConnectionId>,
     pending_events: VecDeque<ToSwarm<PeerReportEvent, Void>>,
+    ips_to_peers: HashMap<IpAddr, HashSet<PeerId>>,
```

Comment:
It seem that this is currently used only to track a number of connection per ip. If we don't want this to be usable for others use cases in the future we could use 
`HashMap<IpAddr, usize>`

It will simplify the current usage that we have by avoiding having to do `.len()` to measure it.